pkg/cluster/delete.go

```diff
@@ -356,6 +357,27 @@ func (m *manager) deleteGateway(ctx context.Context) error {
 	return nil
 }
 
+func (m *manager) deleteClusterMsiCertificate(ctx context.Context) error {
+	// The cluster MSI may have been deleted prior to cluster deletion. If that's the case
+	// we will have already deleted the certificate.
+	if m.doc.OpenShiftCluster.Identity == nil || m.doc.OpenShiftCluster.Identity.UserAssignedIdentities == nil || len(m.doc.OpenShiftCluster.Identity.UserAssignedIdentities) == 0 {
```

Comment:
So this is an example of the previous logic I said might want to be functionized. Especially so it's easier to mentally parse.

Reply:
Closing this thread out to address this idea in your other thread about it.